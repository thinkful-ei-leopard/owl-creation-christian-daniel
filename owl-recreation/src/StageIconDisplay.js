import React from 'react';
import participants from './STORE';
import ReactDOM from 'react-dom';
import StageIcon from './StageIcon';

function StageIconDisplay(props) {
    return (
        <div className='stage-display'>
            {props.participants.map((participant) => 
                <StageIcon 
                    key={participant.id}
                    avatar={participant.avatar}
                    onStage={participant.onStage}
                />
            )}
        </div>
    )
}

export default StageIconDisplay;