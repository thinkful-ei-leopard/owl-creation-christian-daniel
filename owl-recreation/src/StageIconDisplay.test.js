import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import StageIconDisplay from './StageIconDisplay';

describe('<StageIconDisplay />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<StageIconDisplay participants={[{id: 1,
            name: 'Koren Templeton',
            avatar:
                'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
            inSession: true,
            onStage: true}, 
            {id: 2,
                name: 'Caty Flucker',
                avatar:
                    'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
                inSession: true,
                onStage: false}]}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<StageIconDisplay  participants={[{id: 1,
                name: 'Koren Templeton',
                avatar:
                    'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
                onStage: true}, 
                {id: 2,
                    name: 'Caty Flucker',
                    avatar:
                        'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
                    onStage: false}]} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});