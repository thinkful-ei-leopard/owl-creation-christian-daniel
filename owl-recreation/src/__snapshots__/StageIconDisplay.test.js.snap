// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StageIconDisplay /> renders the UI as expected 1`] = `
<div
  className="stage-display"
>
  <div
    className="stageIcon"
  >
    <img
      alt="stage icon"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <p />
    <div
      className="onStage"
    />
  </div>
  <div
    className="stageIcon"
  >
    <img
      alt="stage icon"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <p />
    <div
      className="notOnStage"
    />
  </div>
</div>
`;
